packages/sdk-codegen/src/codeGen.ts

```diff
@@ -235,9 +239,9 @@ export abstract class CodeGen implements ICodeGen {
         )
         propertyValues = props.join(this.enumDelimiter)
       } else {
-        Object.values(type.properties).forEach((prop) =>
-          props.push(this.declareProperty(bump, prop))
-        )
+        Object.values(type.properties)
+          .sort(this.sortProperties)
```

Comment:
what's this sort call for?

Reply:
I should have just closed this PR till it was ready to review. I was going to go through and annotate the PR with why I made certain changes, including this one:

In python I reverted to having all classes define their own `__init__` for consistency and readability. That exposed a bug:

when relying on `kw_only=True` argument to the `attr.s()` class decorator, the order of required vs optional property declarations didn't matter. But now that we're not using that argument, and instead passing `init=False` - `attr` blows up saying that "optional declarations cannot come before required ones". So, in order for the python sdk to "compile" we need to make sure that all required properties are bubbled up to the top.